Add plugin: Kanban Status Updater

ankit-kapur opened the pull request:
# I am submitting a new Community Plugin

## Repo URL
https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin

## Release Checklist
  * [x]  I have tested the plugin on
    * [x]   Windows
    * [x]   macOS
    * [x]   Linux
    * [ ]   Android _(if applicable)_
    * [ ]   iOS _(if applicable)_
  * [x]  My GitHub release contains all required files
    * [x]  `main.js`
    * [x]  `manifest.json`
    * [x]  `styles.css` _(optional)_
  * [x]  GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
  * [x]  The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
  * [x]  My README.md describes the plugin's purpose and provides clear usage instructions.
  * [x]  I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
  * [x]  I have added a license in the LICENSE file.
  * [x]  My project respects and is compatible with the original license of any code from other plugins that I'm using.  I have given proper attribution to these other projects in my `README.md`.

ankit-kapur commented:
Addressed comments.

ankit-kapur commented:
is this ok to merge?

joethei commented:
[// Add status bar item](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L50-L53)
The status bar serves no real purpose in your plugin, remove this code and the other code referencing the status bar.

[const content = await this.app.vault.read(file);](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L409-L449), [](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L469-L474)
Use the `processFrontMatter` API to edit note frontmatter. (requires `minAppVersion` to be 1.1.0)

[containerEl.createEl('h2', {text: 'Kanban Status Updater Settings'});](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L535)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[.setName('Status Property Name')](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L538), [.setName('Show Notifications')](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L549), [.setName('Debug Mode')](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L559), [.setName('Test Plugin')](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L576)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[// Performance info](https://github.com/ankit-kapur/obsidian-kanban-status-updater-plugin/blob/fb5acef605f86de6e27b61346ae7ef620cf8f5ee/main.ts#L584-L606)
This should not be part of the settings tab, but rather the README

ankit-kapur commented:
@joethei thank you for taking a look, addressed your comments. 
- Removed the top-level heading from the settings tab:
- Updated all setting names to use sentence case instead of title case:
- Changed ux names to sentence case
- Removed the Performance Optimization and Troubleshooting sections from the settings tab
- Removed all code related to these sections from the settings display function
- Updated the README.md file for consistency:
- Updated the setting names in the Configuration section to use sentence case
- Updated the minimum app version requirement from v0.15.0 to v1.1.0 to align with the changes made to manifest.json
